### DISCUSSION TITLE
If I want to use websocket in my project, whether there need two gateways?

### DISCUSSION THREAD

**Discussion by liaodalin19903** [moleculerjs/moleculer#1190] [Q&A] [answered]
In my requirement, I will use both normal HTTP request and websocket request in my project, 
whether in moleculer I will use two Gateways? 

such as, [moleculer-web](https://github.com/moleculerjs/moleculer-web#readme) + [moleculer-socketio](https://github.com/davidroman0O/moleculer-socketio) ?

like below:

![image](https://user-images.githubusercontent.com/12727582/223406131-45751d81-9275-4663-b43a-f6271ec91810.png)

**Comment by icebob** [comment, marked as answer]
First of all, the [moleculer-io](https://github.com/moleculerjs/moleculer-io) is the official Socket.io library.

You can use the moleculer-web HTTP server as the server of the moleculer-io as well, so no need to start two HTTP server.
More info: https://github.com/moleculerjs/moleculer-io#init-server
Or real example: https://github.com/icebob/kantab/blob/master/backend/services/api.service.js
